presentation/testing/src/presentation-testing/IModelUtilities.ts

```diff
@@ -6,84 +6,89 @@
  * @module IModel
  */
 
-import { IModelDb, IModelJsFs, InformationPartitionElement, SnapshotDb } from "@itwin/core-backend";
+import { IModelDb, IModelJsFs, SnapshotDb } from "@itwin/core-backend";
 import { Id64String } from "@itwin/core-bentley";
-import { Code, CodeScopeProps, CodeSpec, ElementAspectProps, ElementProps, IModel, LocalFileName, ModelProps } from "@itwin/core-common";
+import { BisCodeSpec, Code, CodeScopeProps, CodeSpec, ElementAspectProps, ElementProps, LocalFileName, ModelProps } from "@itwin/core-common";
 import { IModelConnection, SnapshotConnection } from "@itwin/core-frontend";
 import path from "path";
-import { TEST_OUTPUT_DIR } from "./Helpers";
+import { getTestOutputDir } from "./Helpers";
 
+/**
+ * Interface for IModel builder pattern. Used for building IModels to test rulesets.
+ *
+ * @public
+ */
 export interface TestIModelBuilder {
   insertModel<TProps extends ModelProps>(props: TProps): Id64String;
   insertElement<TProps extends ElementProps>(props: TProps): Id64String;
   insertAspect<TProps extends ElementAspectProps>(props: TProps): void;
-  createCode(scopeModelId: CodeScopeProps, codeSpecName: string, codeValue: string): Code;
-  getUniqueModelCode(newModelCodeBase: string): Code;
+  /** codeValue has to be unique */
+  createCode(scopeModelId: CodeScopeProps, codeSpecName: BisCodeSpec, codeValue: string): Code;
 }
 
+/**
+ * Function that takes builder actions and returns a connection to the build database.
+ * @param name Name of test IModel
+ * @param cb Callback function that executes all given builder actions
+ *
+ * @public
+ */
 export async function buildTestIModel(name: string, cb: (builder: TestIModelBuilder) => void): Promise<IModelConnection> {
-  const outputFile = IModelBuilder.prepareOutputFile(name);
+  const outputFile = setupOutputFileLocation(name);
   const db = SnapshotDb.createEmpty(outputFile, { rootSubject: { name } });
   const builder: TestIModelBuilder = new IModelBuilder(db);
   try {
     cb(builder);
   } finally {
+    db.saveChanges("Created test IModel");
     db.close();
   }
   return SnapshotConnection.openFile(outputFile);
 }
 
+/**
+ * Default implementation of IModel builder pattern. Used for building IModels to test rulesets.
+ *
+ * @public
```

Comment:
I don't think we're planning this to be used by our consumers directly, right? If that's the case, it should be marked `@internal`.

Reply:
Marked all added exports as internal

Reply:
`getTestOutputDir` is exported and doesn't have a release tag

Reply:
Added `@beta`